packages/gatsby/src/utils/graphql-typegen/file-writes.ts

```diff
@@ -0,0 +1,88 @@
+import * as fs from "fs-extra"
+import { join } from "path"
+import { slash } from "gatsby-core-utils"
+import { printSchema } from "graphql"
+import { Store, AnyAction } from "redux"
+import reporter from "gatsby-cli/lib/reporter"
+import type { IGatsbyState, IStateProgram } from "../../redux/types"
+import { stabilizeSchema } from "./utils"
+
+const OUTPUT_PATHS = {
+  schema: `.cache/typegen/schema.graphql`,
+  fragments: `.cache/typegen/fragments.graphql`,
+  config: `.cache/typegen/graphql.config.json`,
+}
+
+export async function writeGraphQLConfig(
+  program: IStateProgram
+): Promise<void> {
+  try {
+    const base = program.directory
+    const configJSONString = JSON.stringify(
+      {
+        schema: OUTPUT_PATHS.schema,
+        documents: [`src/**/**.{ts,js,tsx,jsx}`, OUTPUT_PATHS.fragments],
+        extensions: {
+          endpoints: {
+            default: {
+              url: `${program.https ? `https://` : `http://`}${program.host}:${
+                program.proxyPort
+              }/___graphql`,
+            },
+          },
+        },
+      },
+      null,
+      2
+    )
+
+    const outputPath = slash(join(base, OUTPUT_PATHS.config))
```

Comment:
No need to use slash. You only need it when storing things in cache or comparing paths.

```suggestion

    const outputPath = join(base, OUTPUT_PATHS.config))
```